packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -109,6 +110,153 @@ typedef Tpetra::global_size_t GST;
 } while (false)
 
 
+
+
+/***********************************************************************************/
+template<class MatrixClass, class MultiVectorClass>
+void localReducedMatvec(const MatrixClass & A_lcl,
+                        const MultiVectorClass & X_lcl,
+                        const int userNumRows,
```

Comment:
`userNumRows` should have the same type as `A_lcl.numRows()`.  If `MatrixClass` is a `KokkosSparse::CrsMatrix`, then you can get the type from the `ordinal_type` typedef in `KokkosSparse::CrsMatrix`.

Reply:
Au contraire!  The point of this kernel is that the user can multiply *fewer* rows that the whole matrix.

Reply:
@csiefer2 Can't you just create a "subview" of the matrix that only views a subset of its rows, then call the existing kernel?